### DISCUSSION TITLE
How to run a second server alongside the nuxt3 server ?

### DISCUSSION THREAD

**Discussion by pand-app** [nuxt/nuxt#17436] [Questions]
Dear community,

I'm trying to migrate [nuxt-prometheus-module](https://gitlab.com/qonfucius/nuxt-prometheus-module) to nuxt3, but I'm out of ideas.

The goal is to provide, on a second port (9091 for example, when nuxt run in 3000), an HTTP endpoint that expose nodeJS information for Prometheus.

Easy to, in a kubernetes environment, to expose 3000 outside and keeping 9091 for internal usage (prometheus collecting).
Thanks to this exposure, we can detect in prod things like memory leak in nodeJS that cause reboot of the pod.

In Nuxt2, I was using the `listen` hook, that not exists in Nuxt3.
Maybe should I hook to Nitro or h3 somewhere, but I don't know how to do that in a module...

Any suggest are welcome.

Best regards,
Gaël

**Comment by Flanker72** [comment]
You may create Nitro plugin which will be applied during the main server start. There you may run your additional server logic.
Check https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82 - in Step 2 I wrote how to compose such a plugin.
